tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/service/websocket/FailoverClientTest.java

```diff
@@ -0,0 +1,152 @@
+            BallerinaTestException {
+        remoteServer15200 = initiateServer(SEVER_PORT);
+        WebSocketTestClient client = initiateClient(URL);
+        sendTextDataAndAssert(client, MESSAGE);
+        closeConnection(client, remoteServer15200);
+    }
+
+    @Test(description = "Tests the failover webSocket client by starting the first server in the target URLs.")
+    public void testBinaryFrameForFailover() throws URISyntaxException, InterruptedException, BallerinaTestException {
+        remoteServer15300 = initiateServer(PORT);
+        WebSocketTestClient client = initiateClient(URL);
+        sendBinaryDataAndAssert(client);
+        closeConnection(client, remoteServer15300);
+    }
+
+    @Test(description = "Tests the failover webSocket client by doesn't start the any server in the targets URLs")
+    public void testFailingFailover() throws URISyntaxException, InterruptedException {
+        CountDownLatch countDownLatch = new CountDownLatch(1);
+        WebSocketTestClient client = initiateClient(URL);
+        client.setCountDownLatch(countDownLatch);
+        countDownLatch.await(TIMEOUT_IN_SECS, TimeUnit.SECONDS);
+        CloseWebSocketFrame closeWebSocketFrame = client.getReceivedCloseFrame();
+        Assert.assertNotNull(closeWebSocketFrame);
+        Assert.assertEquals(closeWebSocketFrame.statusCode(), 1011);
+        Assert.assertTrue(closeWebSocketFrame.reasonText().contains("Unexpected condition"));
+        closeWebSocketFrame.release();
+    }
+
+    @Test(description = "Tests the failover webSocket client by starting the both server in the target URLs.")
+    public void testFailoverWithBothServer() throws URISyntaxException, InterruptedException,
+            BallerinaTestException {
+        remoteServer15200 = initiateServer(SEVER_PORT);
+        remoteServer15300 = initiateServer(PORT);
+        WebSocketTestClient client = initiateClient(URL);
+        sendTextDataAndAssert(client, MESSAGE);
+        remoteServer15300.stop();
+        sendTextDataAndAssert(client, text);
+        sendBinaryDataAndAssert(client);
+        closeConnection(client, remoteServer15200);
+    }
+
+    @Test(description = "Tests the failover webSocket client by starting the both server in the target URLs.")
+    public void testFailoverWithBothServerFirstStartSecondServer() throws URISyntaxException, InterruptedException,
+            BallerinaTestException {
+        remoteServer15200 = initiateServer(SEVER_PORT);
+        WebSocketTestClient client = initiateClient(URL);
+        sendTextDataAndAssert(client, MESSAGE);
+        remoteServer15300 = initiateServer(PORT);
+        remoteServer15200.stop();
+        sendTextDataAndAssert(client, text);
+        closeConnection(client, remoteServer15300);
+    }
+
+    @Test(description = "Tests handshake's waiting time exception")
+    public void testCountDownLatch() throws URISyntaxException, InterruptedException, BallerinaTestException {
```

Comment:
Shall we have a better name? Does this test really test a count down latch?
```suggestion
    public void testHandshakeTimout() throws URISyntaxException, InterruptedException, BallerinaTestException {
```